core/domain/story_domain.py

```diff
@@ -302,11 +302,11 @@ def get_incremented_node_id(cls, node_id: str) -> str:
         return incremented_node_id
 
     @classmethod
-    def require_valid_node_id(cls, node_id: str) -> None:
+    def require_valid_node_id(cls, node_id: Optional[str]) -> None:
```

Comment:
I don't think it makes sense to pass None as a node_id here.

Shouldn't this be handled by the caller? In particular, why is someone calling get_ordered_nodes() if there are no nodes in the story? That seems to be more what the problem actually is and I would expect the fix to be in that area instead.

Reply:
Talked to Kevin offline—we think it would also make sense if `get_ordered_nodes()` returned an empty list if there are no nodes.

Reply:
Yup I agree with that last statement, I think that makes sense (though just double-check where it's used to make sure that that assumption works for the callers).

We still don't need to relax the requirement for require_valid_node_id(), though, right? In general, it seems odd to me to pass None for an argument labelled node_id, since None is clearly not a node_id (this principle applies elsewhere too).